### PR TITLE
Fix the Node and pnpm setup actions

### PR DESCRIPTION
**Pull request by whitphx**
# Description

* Update `pnpm/action-setup` because the older versions emit deprecation warnings such as "The `save-state` command is deprecated and will be disabled soon.".
  * This has been fixed with 2.2.4: https://github.com/pnpm/action-setup/releases/tag/v2.2.4
* Set the NodeJS version as 18 because `client/js` requires `node>=18` (https://github.com/gradio-app/gradio/blob/2a30deed8429716c10cd184c44e9b6a0e02b5d47/client/js/package.json#L29-L31) and it shows a warning like below.
```
client/js                                |  WARN  Unsupported engine: wanted: {"node":">=18.0.0"} (current: {"node":"v16.20.0","pnpm":"7.33.0"})
```

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: .github/workflows/ui.yml [hunk 1]
```diff
@@ -21,12 +21,12 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v3
-      - uses: pnpm/action-setup@v2.2.1
+      - uses: pnpm/action-setup@v2.2.4
```
Comment (pngwn) [inline .github/workflows/ui.yml:24]:
```suggestion
      - uses: pnpm/action-setup@v2
```

File: .github/workflows/deploy-pypi.yml [hunk 2]
```diff
@@ -33,7 +33,7 @@ jobs:
     - name: Check new pypi version
       run: python scripts/check_pypi.py
     - name: Install pnpm
-      uses: pnpm/action-setup@v2.2.2
+      uses: pnpm/action-setup@v2.2.4
```
Comment (pngwn) [inline .github/workflows/deploy-pypi.yml:36]:
```suggestion
      uses: pnpm/action-setup@v2
```

File: .github/workflows/build-pr.yml [hunk 3]
```diff
@@ -15,12 +15,12 @@ jobs:
       with:
         python-version: '3.9'
     - name: Install pnpm
-      uses: pnpm/action-setup@v2.2.2
+      uses: pnpm/action-setup@v2.2.4
```
Comment (pngwn) [inline .github/workflows/build-pr.yml:18]:
```suggestion
      uses: pnpm/action-setup@v2
```

File: .github/workflows/backend.yml [hunk 4]
```diff
@@ -91,12 +91,12 @@ jobs:
           test/requirements-37.txt
           test/requirements.txt
     - name: Install pnpm
-      uses: pnpm/action-setup@v2.2.2
+      uses: pnpm/action-setup@v2.2.4
```
Comment (pngwn) [inline .github/workflows/backend.yml:94]:
```suggestion
      uses: pnpm/action-setup@v2
```

File: .github/workflows/ui.yml [hunk 5]
```diff
@@ -45,12 +45,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: pnpm/action-setup@v2.2.1
+      - uses: pnpm/action-setup@v2.2.4
```
Comment (pngwn) [inline .github/workflows/ui.yml:48]:
```suggestion
      - uses: pnpm/action-setup@v2
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -19,6 +19,7 @@
 ## Other Changes:
 
 - Performance optimization in the frontend's Blocks code by [@akx](https://github.com/akx) in [PR 4334](https://github.com/gradio-app/gradio/pull/4334)
+- Maintain the GitHub Actions workflows by [@whitphx](https://github.com/whitphx) in [PR 4394](https://github.com/gradio-app/gradio/pull/4394)
```
Comment (pngwn) [inline CHANGELOG.md:22]: This doesn't need a change log entry as we aren't changing the gradio code.

**Review by pngwn — CHANGES_REQUESTED** [review]

**Comment by whitphx** [comment]
Looks like the CI failed due to the Spaces errors

**Comment by whitphx** [comment]
Thanks!

**Pull request merged**